src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todoId: number
+  setTodoId: (currentTodoId: number | null) => void
+  todo: Todo
+}
+
+export const TodoModal: React.FC<Props> = ({ todoId, setTodoId, todo }) => {
+  const [user, setUser] = useState<null | User>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (todoId !== null) {
+      setLoading(true);
+
+      getUser(todo.userId)
+        .then(setUser)
+        .finally(() => setLoading(false));
+    }
+  }, [todo, todoId]);
```

Comment:
your modal will disappear when you close it and render again when you select another todo. It means that useEffect will fire on initial render every time. Therefore we can leave an empty array here